analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -466,5 +466,97 @@ public void FlowCapture_SetsCapture()
+int[,] arrMulti = new int[2, 3];
+int[][] arrJagged = new int[2][];
+
+Tag(""arr1"", arr1);
+Tag(""arr2"", arr2);
+Tag(""arr3"", arr3);
+Tag(""arrMulti"", arrMulti);
+Tag(""arrJagged"", arrJagged);";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.ArrayCreation);
+            validator.ValidateTag("arr1", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("arr2", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("arr3", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("arrMulti", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("arrJagged", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void DelegateCreation_SetsNotNull()
+        {
+            const string code = @"
+var pointer = Main; // Delegate creation to encapsulating method
+Tag(""pointer"", pointer);";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.DelegateCreation);
+            validator.ValidateTag("pointer", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void DynamicObjectCreation_SetsNotNull()
+        {
+            const string code = @"
+var s = new Sample(dynamicArg);
+Tag(""s"", s);";
+            var validator = SETestContext.CreateCS(code, ", dynamic dynamicArg").Validator;
+            validator.ValidateContainsOperation(OperationKind.DynamicObjectCreation);
+            validator.ValidateTag("s", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void ObjectCreation_SetsNotNull()
+        {
+            const string code = @"
+object assigned;
+var obj = new Object();
+var declared = new Exception();
+assigned = new EventArgs();
+
+Tag(""declared"", declared);
+Tag(""assigned"", assigned);
+Tag(""object"", obj);";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.ObjectCreation);
+            validator.ValidateTag("declared", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("assigned", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("object", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void TypeParameterObjectCreation_SetsNotNull()
+        {
+            const string code = @"
+public void Main<T>() where T : new()
+{
+    var value = new T();
+    Tag(""value"", value);
+}
+
+private void Tag(string name, object arg) { }";
```

Comment:
Maybe better:
```suggestion
private void Tag(string name, T arg) { }";
```

Reply:
That's irrelevant. The `Tag` is only scaffolding noise that is not available when `CreateCSMethod` is called, contrary to `CreateCS` that provides it.